### DISCUSSION TITLE
Texture compression results in bigger file

### DISCUSSION THREAD

**Discussion by mboegl** [zeux/meshoptimizer#659] [Q&A] [answered]
Hi!

When using only the `-tc` flag with a `253Mb` big glb-file, the result is somehow `281Mb`. Any ideas on why or what to do different?

**Comment by zeux** [comment, marked as answer]
I believe that's possible if the source file contains JPEG textures. You can try using lower `-tq` settings (default is 8, so eg try `-tq 5`), but overall Basis Universal doesn't guarantee that the file size is going to be smaller compared to lossy alternatives like JPEG - however, your file will consume dramatically less video memory when loaded, so it's a tradeoff.

**Reply by mboegl** [reply]
Thank you, that makes sense! However, I don't understand what you mean with

> your file will consume dramatically less video memory when loaded

Even though the file size is bigger, it will consume less video memory? Would you mind explaining that briefly?

**Reply by zeux** [reply]
Briefly, JPG/PNG are compressing image formats on disk, but the GPUs can't directly decode these for display purposes, so the browser loads the file from disk/network and then decompresses it into ~4 bytes per pixel (+30% for mipmaps). When using `-tc`, gltfpack uses BasisU super-compression - see https://www.khronos.org/news/press/khronos-ktx-2-0-textures-enable-compact-visually-rich-gltf-3d-assets - which is still compressed on disk/network, and still is decompressed for display purposes, but it's decompressed into a special block-compressed format that GPUs can read directly, which takes ~0.5-1 bytes per pixel (+30% for mipmaps). https://www.khronos.org/ktx/ has an image with an example of the difference between file size & in-memory size between different formats for one model, but of course the actual comparison is going to depend on the formats and image data used.
